src/reference/informative/informC04_component_reference.rst

```diff
@@ -8,4 +8,189 @@
+
+      .. code-block:: xml
+
+        <model name="SurbitonNeighbourhood">
+          ...  
+          <component name="WomenInTheNeighbourhood" />
+          <component name="MenInTheNeighbourhood" />
+
+          <!-- Throughout the importing model (i.e.: this model), the imported
+              items are referred to by their local name attribute ("BarbaraGood" etc), 
+              not the name they are called within their imported family units ("wife"). -->
+          <encapsulation>
+            <component_ref component="WomenInTheNeighbourhood">
+              <component_ref component="BarbaraGood" />
+              <component_ref component="MargotLeadbetter" />
+            </component_ref>
+            <component_ref component="MenInTheNeighbourhood">
+              <component_ref component="TomGood" />
+              <component_ref component="JerryLeadbetter" />
+            </component_ref>
+          </encapsulation>
+        </model>
+
+    This particular encapsulation structure means that the women (Barbara and Margot) are essentially unable to have any contact with the men (Tom and Jerry) even though their original components in the models from which they were imported *were* able to access one another.
+
+    Note also that these locality naming-calling rules are be applied through multiple generations of importing.
+    Since The Good Life is a TV show, there are actors who play the roles of each of the characters. 
+    This could be reflected by using another generation of imports within the two family files like this:
+
+    .. code::
+
+        model: SurbitonNeighbourhood
+          ├─ component: WomenInTheNeighbourhood
+          │   ├─ component: BarbaraGood <╴╴╴╴╴╴╴╴╴╴╴╴╴┐
+          │   └─ component: MargotLeadbetter <╴╴╴╴╴╴┐ ╷
+          └─ component: MenInTheNeighbourhood       ╷ ╷
+              ├─ component: TomGood <╴╴╴╴╴╴╴╴╴╴╴╴╴┐ ╷ ╷
+              └─ component: JerryLeadbetter <╴╴╴┐ ╷ ╷ ╷
+                                                ╷ ╷ ╷ ╷
+                                      imported components
+                                                ╵ ╵ ╵ ╵
+                # In LeadbetterFamily.cellml:   ╵ ╵ ╵ ╵
+                model: TheLeadbetterFamilyUnit  ╵ ╵ ╵ ╵
+        ┌╴╴╴╴╴╴╴╴> ├─ component: husband ╴╴╴╴╴╴╴┘ ╵ ╵ ╵
+        ╷ ┌ ╴╴╴╴╴> └─ component: wife ╴╴╴╴╴╴╴╴╴╴╴╴┼╴┘ ╵
+        ╷ ╷                                       ╵   ╵
+        ╷ ╷     # In GoodFamily.cellml:           ╵   ╵
+        ╷ ╷     model: TheGoodFamilyUnit          ╵   ╵
+        ╷ ╷ ┌╴╴╴> ├─ component: husband ╴╴╴╴╴╴╴╴╴╴┘   ╵
+        ╷ ╷ ╷ ┌╴> └─ component: wife ╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴┘
+        ╷ ╷ ╷ ╷
+      imported components
+        ╵ ╵ ╵ ╵
+        ╵ ╵ ╵ ╵  # In CastOfCharacters.cellml:
+        ╵ ╵ ╵ ╵  model: 
+        ╵ ╵ ╵ └╴╴╴├─ component: FelicityKendal
+        ╵ ╵ └╴╴╴╴╴├─ component: RichardBriers
+        ╵ └╴╴╴╴╴╴╴├─ component: PenelopeKeith
+        └╴╴╴╴╴╴╴╴╴└─ component: PaulEddington
+
+    .. code-block:: xml
```

Comment:
Shouldn't we have the following before (i.e. the same as for the other CellML syntax bits):

```
    .. container:: toggle

      .. container:: header

        See CellML syntax
```

Reply:
I haven't done the toggling for where there are just short sections of CellML, or where the syntax itself is the important thing.  I guess I figured this was "short enough" ... no strong feelings about it though, I can add if you'd prefer.

Reply:
For consistency, I would add a way to toggle the code. It looked "strange" to me to see some code here while earlier I had to toggle similar bits of code.

Reply:
Done in bd6d599.